### DISCUSSION TITLE
Is there any drawback to using the default logger?

### DISCUSSION THREAD

**Discussion by bobgeller401** [gabime/spdlog#2130] [Q&A] [answered]
Hi, 

Thanks for the great library! Just wondering, is there any drawback to using the default logger except that the logger stays alive until the program exits? RAII/controlling the scope seems the only advantage of creating your own logger. Thanks.

Cheers,
Bob

**Comment by gabime** [comment, marked as answer]
Hi,

Thanks for the nice feedback!

No drawback except maybe a slight performance penalty since spdlog needs to fetch the logger from the registry in each call: https://github.com/gabime/spdlog/blob/8826011c81b69e1f8427be520b357c19c74ea480/include/spdlog/spdlog-inl.h#L117

**Reply by bobgeller401** [reply]
Thanks a lot for the quick response! Makes sense!
